Device-omiauth/app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,20 @@
+class Users::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    @user = User.find_for_google_oauth2(request.env['omniauth.auth'],
+      current_user)
+
+    if @user.persisted?
+      flash[:notice] = I18n.t 'devise.omniauth_callbacks.success',
+        kind: 'Google'
+      sign_in_and_redirect @user, event: :authentication
+    else
+      session['devise.google_data'] = request.env['omniauth.auth']
```

Comment:
This case will never happen, because the user is always persisted when you got from first_or_create